api/main_endpoints/routes/Messages.js

```diff
@@ -0,0 +1,113 @@
+const {
+  UNAUTHORIZED,
+  BAD_REQUEST,
+  SERVER_ERROR
+} = require('../../util/constants').STATUS_CODES;
+const express = require('express');
+const router = express.Router();
+const bodyParser = require('body-parser');
+const User = require('../models/User.js');
+const logger = require('../../util/logger');
+
+router.use(bodyParser.json());
+
+const clients = {};
+
+const writeMessage = ((roomId, message) => {
+  if (clients[roomId]) {
+    clients[roomId].forEach(res => res.write(`data: ${JSON.stringify(message)}\n\n`));
+  }
+});
+
+router.post('/send', async (req, res) => {
+
+  const {apiKey, message, id} = req.body;
+
+  const required = [
+    {value: apiKey, title: 'API Key', },
+    {value: message, title: 'Message', },
+    {value: id, title: 'Room ID', },
+  ];
+
+  const missingValue = required.find(({value}) => !value);
+
+  if (missingValue){
+    res.status(BAD_REQUEST).send(`You must specify a ${missingValue.title}`);
+    return;
+  }
+
+  try {
+    User.findOne({apiKey}, (error, result) => {
+      if (error) {
+        logger.error('/send received an invalid API key: ', error);
+        res.sendStatus(SERVER_ERROR);
+        return;
+      }
+      if (result) {
+        writeMessage(id, message);
+        return res.json({status: 'Message sent'});
+      }
+      return res.sendStatus(UNAUTHORIZED);
+    });
+  } catch (error) {
+    logger.error('Error in /send: ', error);
+    res.sendStatus(SERVER_ERROR);
+  }
+});
+
+router.get('/listen', async (req, res) => {
+  const {apiKey, id} = req.query;
+
+  const required = [
+    {value: apiKey, title: 'API Key', },
+    {value: id, title: 'Room ID', }
+  ];
+
+  const missingValue = required.find(({value}) => !value);
+
+  if (missingValue){
+    res.status(BAD_REQUEST).send(`You must specify a ${missingValue.title}`);
+    return;
+  }
+
+  try {
+    User.findOne({apiKey}, (error, result) => {
+      if (error) {
+        logger.error('/listen received an invalid API key: ', error);
+        res.sendStatus(SERVER_ERROR);
+        return;
+      }
+      if (result) { // logic to keep the connection alive
```

Comment:
lets invert the logic for less indentation

```js
if (!result) {
  return res.sendStatus(UNAUTHORIZED);
}
// continue your code below, no need for an `else`
const headers = {
          'Content-Type': 'text/event-stream',
          'Connection': 'keep-alive',
          'Cache-Control': 'no-cache',
};
// ...
```

Reply:
done